underfs/s3/src/main/java/alluxio/underfs/s3/S3UnderFileSystem.java

```diff
@@ -326,6 +326,25 @@ public InputStream open(String path) throws IOException {
     }
   }
 
+  /**
+   * Opens a S3 object at given position and returns the opened input stream.
+   *
+   * @param path the S3 object path
+   * @param pos the position to open at
+   * @return the opened input stream
+   * @throws IOException if failed to open file at position
+   */
+  public InputStream openAtPosition(String path, long pos) throws IOException {
+    try {
+      path = stripPrefixIfPresent(path);
+      S3Object obj = mClient.getObject(mBucketName, path, null, null, null, null, pos, null);
+      return obj.getDataInputStream();
```

Comment:
Could we make a new constructor in `S3InputStream` instead of having some stream logic here?